src/core/utils.ts

```diff
@@ -5,6 +5,7 @@ import path from "path";
 import YAML from "yaml";
 import { DEFAULT_CONFIG } from "../config";
 import { detectRuntime, RuntimeType } from "./runtimes";
+const { readdir, readFile } = require("fs").promises;
```

Comment:
Everywhere else we've used the sync methods for working with `fs`, so why change it here? Also, I know it's more combersome, but I'd still use `import` but just do a second destructuring call, since you can't do nested destructuring on `import`:

```js
import { promises as fsPromises } from "fs";

const { readdir, readFile } = fsPromises;
```

Reply:
I think that using the sync calls was a mistake from the start. I think we still have time to change that.